panic found on S3Select

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
No panics
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
```
panic: "POST /testbucket/test/testSelectEmptyFile?select&select-type=2": runtime error: invalid memory address or nil pointer dereference
goroutine 612627 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x65
github.com/minio/minio/cmd.setCriticalErrorHandler.func1.1()
	github.com/minio/minio/cmd/generic-handlers.go:468 +0x8e
panic({0x29560c0, 0x694aec0})
	runtime/panic.go:1038 +0x215
github.com/minio/minio/internal/s3select.(*S3Select).Close(0xc003e14848)
	github.com/minio/minio/internal/s3select/select.go:580 +0x22
github.com/minio/minio/cmd.objectAPIHandlers.SelectObjectContentHandler({0x4c370e0, 0x4c370d0}, {0x5029ac0, 0xc01407f5e0}, 0xc014779400)
	github.com/minio/minio/cmd/object-handlers.go:275 +0x16aa
net/http.HandlerFunc.ServeHTTP(...)
	net/http/server.go:2046
github.com/minio/minio/cmd.httpTraceHdrs.func1({0x5029ac0, 0xc01407f5e0}, 0xc0293af1c0)
	github.com/minio/minio/cmd/handler-utils.go:372 +0x53
net/http.HandlerFunc.ServeHTTP(0x2848200, {0x5029ac0, 0xc01407f5e0}, 0x4)
	net/http/server.go:2046 +0x2f
github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0x5029ac0, 0xc01407f5e0}, 0x203005)
	github.com/klauspost/compress@v1.13.6/gzhttp/gzip.go:387 +0x328
net/http.HandlerFunc.ServeHTTP(...)
	net/http/server.go:2046
github.com/minio/minio/cmd.maxClients.func1({0x5029ac0, 0xc01407f5e0}, 0xc014779400)
	github.com/minio/minio/cmd/handler-api.go:252 +0x41c
net/http.HandlerFunc.ServeHTTP(...)
	net/http/server.go:2046
github.com/minio/minio/cmd.collectAPIStats.func1({0x5029ac0, 0xc01407f540}, 0x0)
	github.com/minio/minio/cmd/handler-utils.go:387 +0x112
net/http.HandlerFunc.ServeHTTP(0x5029a60, {0x5029ac0, 0xc01407f540}, 0x10)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.addCustomHeaders.func1({0x5029a60, 0xc0211d0630}, 0x4c37078)
	github.com/minio/minio/cmd/generic-handlers.go:451 +0x3c2
net/http.HandlerFunc.ServeHTTP(0x2c19c80, {0x5029a60, 0xc0211d0630}, 0x39)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setRequestValidityHandler.func1({0x5029a60, 0xc0211d0630}, 0xc014779400)
	github.com/minio/minio/cmd/generic-handlers.go:365 +0x90d
net/http.HandlerFunc.ServeHTTP(0x7fcd63cfe300, {0x5029a60, 0xc0211d0630}, 0x7fcd8d448f18)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setHTTPStatsHandler.func1({0x5035460, 0xc014a90d20}, 0xc014779400)
	github.com/minio/minio/cmd/generic-handlers.go:279 +0x10e
net/http.HandlerFunc.ServeHTTP(0xc0293af200, {0x5035460, 0xc014a90d20}, 0xc01504f348)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setRequestLimitHandler.func1({0x5035460, 0xc014a90d20}, 0xc014779400)
	github.com/minio/minio/cmd/generic-handlers.go:111 +0x38f
net/http.HandlerFunc.ServeHTTP(0xa6, {0x5035460, 0xc014a90d20}, 0x23cdf76)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setCrossDomainPolicy.func1({0x5035460, 0xc014a90d20}, 0x6)
	github.com/minio/minio/cmd/crossdomain-xml-handler.go:43 +0xef
net/http.HandlerFunc.ServeHTTP(0xc014779400, {0x5035460, 0xc014a90d20}, 0x1)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setBrowserRedirectHandler.func1({0x5035460, 0xc014a90d20}, 0xc014779400)
	github.com/minio/minio/cmd/generic-handlers.go:140 +0x183
net/http.HandlerFunc.ServeHTTP(0x0, {0x5035460, 0xc014a90d20}, 0x355a1a92)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setAuthHandler.func1({0x5035460, 0xc014a90d20}, 0xc014779400)
	github.com/minio/minio/cmd/auth-handler.go:512 +0x3d8
net/http.HandlerFunc.ServeHTTP(0xc014779300, {0x5035460, 0xc014a90d20}, 0x1)
	net/http/server.go:2046 +0x2f
github.com/gorilla/mux.(*Router).ServeHTTP(0xc00036c900, {0x5035460, 0xc014a90d20}, 0xc014779200)
	github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
github.com/rs/cors.(*Cors).Handler.func1({0x5035460, 0xc014a90d20}, 0xc014779200)
	github.com/rs/cors@v1.7.0/cors.go:219 +0x1bd
net/http.HandlerFunc.ServeHTTP(0x2b810e0, {0x5035460, 0xc014a90d20}, 0xc0016e49f8)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/cmd.setCriticalErrorHandler.func1({0x5035460, 0xc014a90d20}, 0x50046e0)
	github.com/minio/minio/cmd/generic-handlers.go:475 +0x83
net/http.HandlerFunc.ServeHTTP(0x0, {0x5035460, 0xc014a90d20}, 0x14)
	net/http/server.go:2046 +0x2f
github.com/minio/minio/internal/http.(*Server).Start.func1({0x5035460, 0xc014a90d20}, 0xc0293af1a0)
	github.com/minio/minio/internal/http/server.go:97 +0x1b6
net/http.HandlerFunc.ServeHTTP(0xc014bce98a, {0x5035460, 0xc014a90d20}, 0x4746b8)
	net/http/server.go:2046 +0x2f
net/http.serverHandler.ServeHTTP({0x5021a60}, {0x5035460, 0xc014a90d20}, 0xc014779200)
	net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc01407f4a0, {0x50545b8, 0xc0012e4420})
	net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve
	net/http/server.go:3033 +0x4e8
```

<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Not sure, reproduced while running `hadoop` integration tests.

<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Not easy, do not know the exact test that reproduces this issue.

## Context
Running hadoop integration tests for hadoop-aws
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Regression
Not sure
<!-- Is this issue a regression? (Yes / No) -->
<!-- If Yes, optionally please include minio version or commit id or PR# that caused this regression, if you have these details. -->

## Your Environment
master branch MinIO - erasure-coded setup 4 drives local setup.

--- a/internal/s3select/select.go
+++ b/internal/s3select/select.go
@@ -369,7 +369,7 @@ func (s3Select *S3Select) Open(getReader func(offset, length int64) (io.ReadClos
 		return err
 	}
 
-	panic(fmt.Errorf("unknown input format '%v'", s3Select.Input.format))
+	return fmt.Errorf("unknown input format '%v'", s3Select.Input.format)
 }
 
 func (s3Select *S3Select) marshal(buf *bytes.Buffer, record sql.Record) error {
@@ -577,6 +577,9 @@ OuterLoop:
 
 // Close - closes opened S3 object.
 func (s3Select *S3Select) Close() error {
+	if s3Select.recordReader == nil {
+		return nil
+	}
 	return s3Select.recordReader.Close()
 }
 

